### PR TITLE
feat: Product Card Theming

### PR DESCRIPTION
**Pull request by sergiofontes**
## What's the purpose of this pull request?

This PR applies the new Theming[1] structure to `ProductCard`: new scoped tokens linked to global ones.

## How does it work?

This PR uses local variables to parameterize the `ProductCard` properties, then connects these scoped tokens to the global ones. It also refactors the component's styling by flattening its structure and using `display: grid` to lay things out (rather than `flex`).

Finally, all component variants were moved to `product-card.scss`. For example, the `tiles.scss` file altered the appearance of `ProductCard` at the parent level. A parent should ideally only set the external geometry[2] of its children, allowing for greater composability and styling independence of the component.

### New global tokens
None.

### New local tokens for `ProductCard`
|Local token|Default value / global token linked|
|-|-|
|`--fs-product-card-padding`                  | `var(--fs-spacing-1) var(--fs-spacing-1) var(--fs-spacing-2)`|
  |`--fs-product-card-gap`                      | `var(--fs-grid-gap-0)`|
  |`--fs-product-card-min-width`                | `14rem`|
  |`--fs-product-card-shadow`                   | `none`|
  |`--fs-product-card-shadow-hover`             | `var(--fs-shadow-hover)`|
  |`--fs-product-card-bkg-color`                | `var(--fs-color-body-bkg)`|
  |`--fs-product-card-bkg-color-hover`          | `var(--fs-product-card-bkg-color)`|
  |`--fs-product-card-bkg-color-focus`          | `var(--fs-product-card-bkg-color-hover)`|
  |`--fs-product-card-img-radius`               | `var(--fs-border-radius-default)`|
  |`--fs-product-card-content-padding`          | `var(--fs-spacing-2) 0 0 0`|
  |`--fs-product-card-border-radius`            | `var(--fs-border-radius-default)`|
  |`--fs-product-card-border-width`             | `var(--fs-border-width-default)`|
  |`--fs-product-card-border-color`             | `transparent`|
  |`--fs-product-card-border-color-hover`       | `var(--fs-border-color-light)`|
  |`--fs-product-card-text-title-color`         | `var(--fs-color-text)`|
  |`--fs-product-card-text-title-size`          | `var(--fs-text-size-base)`|
  |`--fs-product-card-text-title-weight`        | `var(--fs-text-weight-regular)`|
  |`--fs-product-card-text-price-color`         | `var(--fs-color-text)`|
  |`--fs-product-card-text-price-size`          | `var(--fs-text-size-base)`|
  |`--fs-product-card-text-price-listing-color` | `var(--fs-color-text-light)`|
  |`--fs-product-card-text-price-listing-size`  | `var(--fs-text-size-legend)`|
  |`--fs-product-card-text-description-color`   | `var(--fs-color-text)`|
  |`--fs-product-card-text-description-size`    | `var(--fs-text-size-legend)`|
  |`--fs-product-card-text-subtitle-color`      | `var(--fs-color-text-light)`|
  |`--fs-product-card-text-subtitle-size`       | `var(--fs-text-size-tiny)`|
  |`--fs-product-card-transition-function`      | `var(--fs-transition-function)`|
  |`--fs-product-card-transition-property`      | `var(--fs-transition-property)`|
  |`--fs-product-card-transition-timing`        | `var(--fs-transition-timing)`|

## How to test it?

The `ProductCard` component should look just like it was before this effort.

## References

[1] Theming architecture: #407 

[2] A parent should change only the following properties of their children:
```scss
.parent > .child {
  position: fixed | absolute | sticky;
  top: any;
  right: any;
  bottom: any;
  left: any;
  margin: any;
  float: any;
  clear: any;
  order: any;
  flex: any;
  grid-column: any;
  grid-row: any;
  grid-area: any;
  width: any;
  height: any;
  box-sizing: content-box | padding-box | inherit;
}
```

### CODE REVIEW

File: src/components/product/ProductCard/ProductCard.tsx [hunk 1]
```diff
@@ -59,10 +59,10 @@ function ProductCard({
 
   return (
     <UICard
-      className="product-card"
-      data-card-variant={variant}
-      data-card-bordered={bordered}
+      data-fs-product-card-variant={variant}
+      data-fs-product-card-bordered={bordered}
       data-card-out-of-stock={outOfStock}
```
Comment (hellofanny) [inline src/components/product/ProductCard/ProductCard.tsx:64]: I think we can remove this data attr, we are not using it anymore and so far we don't need it here.

image







```suggestion
```

File: src/components/product/ProductCard/product-card.scss [hunk 2]
```diff
@@ -84,15 +105,107 @@
...
+  border: var(--fs-border-width-default) solid var(--fs-border-color-light);
+}
+
+// --------------------------------------------------------
+// Template Variants
+// --------------------------------------------------------
+
+[data-fs-product-card-variant="wide"] {
+  --fs-product-card-padding: 0;
+  --fs-product-card-content-padding: var(--fs-spacing-2);
```
Comment (hellofanny) [inline src/components/product/ProductCard/product-card.scss:157]:
```suggestion
  --fs-product-card-content-padding: var(--fs-spacing-3);
```

|Before|
|-|
before-space-3

|After|
|-|
before-space-2

Not sure. Should we keep it or change it?

**Comment by hellofanny** [comment]
I liked the approach of using `display: grid`  instead of flex for this component.

Just to point out some improvements/adjusts that we made:

|Before - base.vtex.app|After|
|-|-|
before|after

- In this new version on mobile, the `badge` is no longer inline, and the images' height has increased.
@renatamottam are you okay with these changes?

Or do you think we can leave `grid-template-rows: auto` for mobile?

https://user-images.githubusercontent.com/3356699/160216477-78944cbf-9aab-4c61-918b-58048e1c2367.mov

**Comment by hellofanny** [comment]
Another improvement from this PR:

|Before - base.vtex.app|After|
|-|-|
bug|!after

- Fixes the empty gaps between the card's image and the card content 🙈

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,6 +52,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `postalCode` from storage to Session context.
 - Updates all tokens naming and simplifies the global styles.
 - Changes `theme.scss` file to `global/tokens.scss`.
+- Applies new local tokens to ProductCard.
```
Comment (filipewl) [inline CHANGELOG.md:55]:
```suggestion
- Applies new local tokens to `ProductCard`.
```

File: src/components/product/ProductCard/product-card.scss [hunk 4]
```diff
@@ -1,78 +1,101 @@
...
-  padding: var(--fs-spacing-1) var(--fs-spacing-1) var(--fs-spacing-2);
-  border: var(--fs-border-width-default) solid transparent;
-  border-radius: var(--fs-border-radius-default);
-  transition: box-shadow .5s ease, border .5s ease;
-
-  &:hover,
-  &:focus {
-    background-color: var(--fs-color-neutral-0);
-  }
+/* stylelint-disable */
```
Comment (filipewl) [inline src/components/product/ProductCard/product-card.scss:3]: What's up with disabling the Stylelint here? Feels like it's going to be easy not to follow its rules later in future changes to this file.

**Review by filipewl — APPROVED** [review]
Looks good to me!

Also noticed the following improvement to stick the price label to the bottom of the card:
Before|After
-|-
Screen Shot 2022-03-28 at 15 14 14|Screen Shot 2022-03-28 at 15 14 16

File: src/components/product/ProductCard/ProductCard.tsx [hunk 5]
```diff
@@ -59,10 +59,10 @@ function ProductCard({
...
+      data-fs-product-card
```
Comment (eduardoformiga) [inline src/components/product/ProductCard/ProductCard.tsx:65]: not so important, but what do you think about placing the data-attr that represents the component as the first?

**Review by eduardoformiga — COMMENTED** [review]
Just a first review so far. 
I'm going to look at the Tiles and other things afterward deeply.

File: src/components/product/ProductCard/product-card.scss (hunk 4 above)
Comment (filipewl) [reply src/components/product/ProductCard/product-card.scss:3]: Better see [this other comment](https://github.com/vtex-sites/base.store/pull/425#discussion_r836595761)!

File: src/components/product/ProductCard/ProductCard.tsx (hunk 5 above)
Comment (sergiofontes) [reply src/components/product/ProductCard/ProductCard.tsx:65]: Sure thing, let's do this.

File: src/components/product/ProductCard/product-card.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergiofontes) [reply src/components/product/ProductCard/product-card.scss:3]: Seems like an option. Could you do it?

**Review by renatamottam — APPROVED** [review]

File: src/components/product/ProductCard/product-card.scss (hunk 2 above)
Comment (sergiofontes) [reply src/components/product/ProductCard/product-card.scss:161]: `var(--fs-spacing-2)` is the correct one.

**Comment by eduardoformiga** [comment]
Hey, finally I have reviewed it deeply now, even after merged, but we can adjust on another PR.
I also like the approach of using the `grid` on the product card 👏 

Some things that I notice:
1. I miss the `line-clamp` for the product card.
2. The `bagde` inline as @hellofanny mentioned previously. Our [Figma](https://www.figma.com/file/cAXtiQGObWxamFVVhviQVM/Handoff-v2-(Homepage)?node-id=925%3A69512) prototype indicates it `inline` in some cases.



I also test the versions using 2 and 4 `Tiles` and looks great!
Screen Shot 2022-03-29 at 18 05 59

Just notice this discrepancy that might be resolved after the `line-clamp`

File: src/components/product/ProductCard/product-card.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/product/ProductCard/product-card.scss:3]: Sure. I'll implement it here on https://github.com/vtex-sites/base.store/pull/430

**Pull request merged**